android/src/app/organicmaps/intent/Factory.java

```diff
@@ -218,6 +211,13 @@ public boolean run(@NonNull MwmActivity target)
           intent.putExtra(MwmActivity.EXTRA_BACK_URL, backUrl);
```

Comment:
That duplicate logic with manually passed back url in Java despite its presence in the parsed C++ API complicates everything. It would be great to refactor this code and query backurl directly from C++, instead of passing it via intent and querying in MWMActivity.onKeyUp